Feature/update readme documentation (#77)
MadBase/MadNet: bridge/README.md

# Madnet/Bridge

This repository contains all solidity smart contracts used by the MadNet. 

## Requirements
* [dapp tools](https://github.com/dapphub/dapptools#installation)
* HardHat (following)

### HardHat

To install hardhat and all requirements to compile and test the smart contracts in this repository,
run the following command at the root of this repository

```bash
$ npm i -g hardhat-shorthand
$ hardhat-completion install
```

## Setup

### Install dependencies 

Install all the necessary dependencies
```bash
$ npm ci
$ npm run init 
```
It is mandatory to run this latest command `npm run init`, every time you run the `clean` command.  

### Running unit tests

You can choose between two different commands to run unit tests. The `test-parallel` will be faster
but you won't see the gas reporter as logged once the tests are running successfully 
`Note: Gas reporting has been skipped because plugin hardhat-gas-reporter does not support the --parallel flag.)`

```bash

> $ npm run test 

Reviewer: What do you think about separating these 2 commands into 2 separated bash blocks? I think it would back you up in the statement that you wrote above about the possible commands and maybe avoiding people blinding copying and pasting and executing the tests twice haha.

$ npm run test-parallel 
```

> ### Deploying

Reviewer: Maybe we should be renaming this to 
```suggestion
### Golang code binding generation
```
We will be adding documentation about the deployment in the near future. We need to finish the hardhat scripts yet.

> Once you reach this point you will now be able to compile and deploy all the contracts. In case you have

Reviewer: ```suggestion
Once you reach this point you will now be able to compile and generate code bindings for all the contracts. In case you have
```

> new contracts **MAKE SURE** to add them in the configuration file `hardha.config.ts` under `"abiExporter" -> "only"` array.

Reviewer: ```suggestion
new contracts **MAKE SURE** to add them in the configuration file `hardhat.config.ts` under `"abiExporter" -> "only"` array.
```

```bash
$ npm run build
```

> 

Reviewer: Can we add a troubleshooting section as we have in the Madnet main repo? For now just include the `npm clean` command in here, if a person is having a problem compiling the contracts, we can add more info as we find necessary in the future.

### Resources

> **Sintax Guidelines** - We follow the standard coding convention for writing solodity code. You can find

Reviewer: Can we put this in a separate section, so it's clear to everyone joining the project? Maybe call it `Solidity Style guide`

the documentation [here](https://docs.soliditylang.org/en/v0.8.9/style-guide.html).

**Formatting** - there is a command, `npm run format` to prettify the codebase based on 
the configuration file `prettier.json`

**Visual Studio Code Plugin** - Group of pluing to make your life easier if you are using 
Visual studio
* Solidity Visual Auditor - [plugin link](https://marketplace.visualstudio.com/items?itemName=tintinweb.solidity-visual-auditor)
* Solidity - [plugin link](https://marketplace.visualstudio.com/items?itemName=JuanBlanco.solidity)
* Remix - [plugin link](https://marketplace.visualstudio.com/items?itemName=RemixProject.ethereum-remix)

> **Diamond Smart Contract pattern** - The main architecture pattern used by the MadNet smart contracts 

Reviewer: Also forgot to mention that we are not using the Diamond pattern anymore, so we can remove it from the docs.
```suggestion
```

> available [here](https://eips.ethereum.org/EIPS/eip-2535).

Reviewer: ```suggestion
```
